pkg/apis/config/v1alpha1/resourceinterpretercustomization_types.go

```diff
@@ -127,25 +138,31 @@ type LocalValueRetention struct {
 type ReplicaResourceRequirement struct {
 	// LuaScript holds the Lua script that is used to discover the resource's
 	// replica as well as resource requirements
+	//
 	// The script should implement a function as follows:
 	//     luaScript: >
 	//         function GetReplicas(desiredObj)
-	//             nodeClaim = {}
-	//             resourceRequest = {}
-	//             result = {}
-	//
-	//             result.replica = desiredObj.spec.replicas
-	//             result.resourceRequest = desiredObj.spec.template.spec.containers[0].resources.limits
+	//             replica = desiredObj.spec.replicas
+	//             requirement = {}
+	//             requirement.nodeClaim = {}
+	//             requirement.nodeClaim.nodeSelector = desiredObj.spec.template.spec.nodeSelector
+	//             requirement.nodeClaim.tolerations = desiredObj.spec.template.spec.tolerations
+	//             requirement.resourceRequest = desiredObj.spec.template.spec.containers[1].resources.limits
+	//             return replica, requirement
+	//         end
 	//
-	//             nodeClaim.nodeSelector = desiredObj.spec.template.spec.nodeSelector
-	//             nodeClaim.tolerations = desiredObj.spec.template.spec.tolerations
-	//             result.nodeClaim = nodeClaim
+	// The content of the LuaScript needs to be a whole function including both
+	// declaration and implementation.
 	//
-	//             return result
-	//         end
+	// The parameters will be supplied by the system:
+	//   - desiredObj: the object represents the configuration to be applied
+	//       to the member cluster.
 	//
-	// LuaScript only holds the function body part, take the `desiredObj` as the function
-	// parameters or global variable, and finally returns the replica number and required resources.
+	// The function expects two return values:
+	//   - replica: the declared replica number
+	//   - requirement: the resource required by each replica expressed with a
```

Comment:
It is recommended to add a type description, such as `replica` is a `number` type and `requirement` is a `table` type.

Reply:
+1

Reply:
Can you give an example?

Reply:
I totally agree with the idea. But after going through the Lua documents about [Values and Types](https://www.lua.org/manual/5.1/manual.html#2.2) I feel hard to describe the types. Maybe the example is clear enough? We can add more descriptions to documents.

Reply:
> We can add more descriptions to documents.

+1